server/grpc/rpc.go

```diff
@@ -275,6 +275,8 @@ func (s *RPC) Init(c context.Context, id string, state rpc.State) error {
 		}
 	}
 
+	s.updateForgeStatus(c, repo, currentPipeline, step)
```

Comment:
This would set the status to pending? Would the changed status in line 297 be send to the forge as well?